docs/design/seccomp.md

```diff
@@ -0,0 +1,286 @@
+seccomp, which is very close to the libseccomp API.  It allows full
+specification of seccomp filters, with arguments, operators, and actions.
+
+Docker allows the specification of a single seccomp filter.  There are
+community requests for:
+
+Issues:
+
+* [docker/22109](https://github.com/docker/docker/issues/22109): composable
+  seccomp filters
+* [docker/21105](https://github.com/docker/docker/issues/22105): custom
+  seccomp filters for builds
+
+#### rkt / appcontainers
+
+The `rkt` runtime delegates to systemd for seccomp support; there is an open
+issue to add support once `appc` supports it.  The `appc` project has an open
+issue to be able to describe seccomp as an isolator in an appc pod.
+
+The systemd seccomp facility is based on a whitelist of system calls that can
+be made, rather than a full filter specification.
+
+Issues:
+
+* [appc/529](https://github.com/appc/spec/issues/529)
+* [rkt/1614](https://github.com/coreos/rkt/issues/1614)
+
+#### HyperContainer
+
+[HyperContainer](https://hypercontainer.io) does not support seccomp.
+
+### Other platforms and seccomp-like capabilities
+
+FreeBSD has a seccomp/capability-like facility called
+[Capsicum](https://www.freebsd.org/cgi/man.cgi?query=capsicum&sektion=4).
+
+#### lxd
+
+[`lxd`](http://www.ubuntu.com/cloud/lxd) constrains containers using a default profile.
+
+Issues:
+
+* [lxd/1084](https://github.com/lxc/lxd/issues/1084): add knobs for seccomp
+
+## Proposed Design
+
+### Seccomp API Resource?
+
+An earlier draft of this proposal described a new global API resource that
+could be used to describe seccomp profiles.  A reasonable decision plane for
+adding a new API resource is functionality that is used by 80% of our users.
+
+After some discussion, it was determined that without a feedback signal from
+users indicating a need to describe new profiles in the Kubernetes API, it is
+not possible to know whether a new API resource is warranted.
+
+That being the case, we will not propose a new API resource at this time.  If
+there is strong community desire for such a resource, we may consider it in
+the future.
+
+Instead of implementing a new API resource, we propose that pods be able to
+reference seccomp profiles by name.
+
+### API changes
+
+We will use annotations to store the names of seccomp profiles to use.  The
+keys will be:
+
+`security.alpha.kubernetes.io/seccomp/container/<container name>`
+
+which will be used to set the seccomp profile of a container, and:
+
+`security.alpha.kubernetes.io/seccomp/pod`
+
+which will set the seccomp profile for the containers of an entire pod.  If a
+pod-level annotation is present, and a container-level annotation present for
+a container, then the container-level profile takes precedence.
+
+The value of these keys will be the formatted name of a seccomp profile:
+
+1.  The name is scoped with a context.  Valid contexts are:
+  1.  `docker/` -- indicates a built-in docker profile, viz: `docker/default` or `docker/unconfined`
```

Comment:
@jonboulle 

I would say that (a) is correct; but I am not sure that I follow (b, c, d).  Let me clarify:

The content of a seccomp profile and how the container runtime handles it is completely opaque to the system.  I think perhaps the content in the proposal currently was skewed unintentionally.  Here's how I think I would restate it (suggesting new copy here, so I may restate a little of this paragraph):

The value of these keys should be container-runtime agnostic. We will establish a format that expresses the conventions for distinguishing between an unconfined profile, the container runtime's default, or a custom profile.  Since format of profile is likely to be runtime dependent, we will consider profiles to be opaque to kubernetes for now.

The following format is scoped as follows:

1.  `runtime/default` - the default profile for the container runtime
2.  `unconfined` - unconfined profile, ie, no seccomp sandboxing
3.  `localhost/<profile-name>` - the profile installed to the node's local seccomp-profile-root

Apologies for the initial phrasing; how does the above strike you?